### PR TITLE
Created --reset-key flag to update OpenAI key

### PR DESCRIPTION
**Pull request by Ismail-Ben**
Resolves: #30 

Added --reset-key flag to update OPENAI_API_KEY in .sgptrc file. Also verified with a dummy chat completion that the newly inputted key works.

Sample output with 2 non-valid keys followed by a real key:
image

### CODE REVIEW

**Comment by Ismail-Ben** [comment]
@TheR1D Any suggested changes to get this in?

**Review by TheR1D — CHANGES_REQUESTED** [review]
I think this option doesn't align with prioritized environment variables in config now, since we can't reset user exported `OPENAI_API_KEY`. But we could add better error handling into `client._request` simply checking respone status code for `403/401` printing the error text and raising `Exit` with error `raise typer.Exit(code=1)`.

**Comment by Ismail-Ben** [comment]
> I think this option doesn't align with prioritized environment variables in config now, since we can't reset user exported `OPENAI_API_KEY`. But we could add better error handling into `client._request` simply checking respone status code for `403/401` printing the error text and raising `Exit` with error `raise typer.Exit(code=1)`.

I've moved the error handling to client._request as suggested. 

As for the reset functionality I added a note to the "help" section of the command as a reminder that the environment variable will still be prioritized. I still believe having a --reset-key option is worthwhile as I would've liked to use it myself when first trying the tool. Instead of having to navigate to the .sgptrc file and update it manually I would've been nice to have this command. 

The config.get() function properly prioritizes the use of the env key if it exists so updating the .sgptrc file wouldn't change this behavior.  In the case that a user runs this command while already having the env variable I highlight in yellow that their env key will still be used.

User updates key with env variable defined:
image

No env variable defined, and fake key:
image

Help section:
image

Let me know what you think @TheR1D . Thanks!

File: sgpt/client.py [hunk 1]
```diff
@@ -58,7 +59,15 @@ def _request(
             timeout=REQUEST_TIMEOUT,
             stream=stream,
         )
-        response.raise_for_status()
+        try:
```
Comment (TheR1D) [inline sgpt/client.py:62]: There is no reason to use `try` block here, we can just check `response.status` above.

File: sgpt/client.py [hunk 2]
```diff
@@ -58,7 +59,15 @@ def _request(
...
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as http_err:
+            # Check if the status code is 403 (Forbidden) or 401 (Unauthorized)
+            if http_err.response.status_code == 403 or http_err.response.status_code == 401:
+                typer.secho("Invalid OpenAI API key.", fg="red")
```
Comment (TheR1D) [inline sgpt/client.py:67]: We can add more info for user, something like `f"Invalid OpenAI API key, update your config file: {cfg.SHELL_GPT_CONFIG_PATH}"`.

File: sgpt/client.py [hunk 3]
```diff
@@ -58,7 +59,15 @@ def _request(
...
+            else:
+                typer.secho(f"HTTPError occurred: {http_err}", fg="red")
```
Comment (TheR1D) [inline sgpt/client.py:69]: We don't need this line since `raise_for_status` should handle it.

**Review by TheR1D — CHANGES_REQUESTED** [review]
It is very common in CLI applications to use runtime configuration file or ENV variables to set API keys (without having dedicated CLI options to reset some of config values). Please remove reset key option, if we show user proper message, it is clear how to reset it.

**Comment by Ismail-Ben** [comment]
Adjusted to reflect @TheR1D 's comments. When using an invalid key the output now looks like:
image

**Review by TheR1D — DISMISSED** [review]
Thank you @Ismail-Ben

**Comment by TheR1D** [comment]
Please check lint issues.

**Comment by TheR1D** [comment]
Thank you, merged.

**Pull request merged**